ui/src/app/subnets/views/SubnetDetails/SubnetDetailsHeader/SubnetActionForms/components/DeleteSubnet/DeleteSubnet.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  Button,
+  Strip,
+  Notification,
+  Row,
+  Col,
+  ActionButton,
+} from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import { useHistory } from "react-router";
+
+import VisuallyHidden from "app/base/components/VisuallyHidden";
+import { useCycled } from "app/base/hooks";
+import { useId } from "app/base/hooks/base";
+import { actions as subnetActions } from "app/store/subnet";
+import { default as subnetSelectors } from "app/store/subnet/selectors";
+import { getCanBeDeleted, getIsDHCPEnabled } from "app/store/subnet/utils";
+import { default as vlanSelectors } from "app/store/vlan/selectors";
+import urls from "app/subnets/urls";
+import type { SubnetActionProps } from "app/subnets/views/SubnetDetails/types";
+import { formatErrors } from "app/utils";
+
+export const DeleteSubnet = ({
+  setActiveForm,
+}: Omit<SubnetActionProps, "activeForm">): JSX.Element | null => {
+  const subnet = useSelector(subnetSelectors.active);
```

Comment:
While we can presume the active subnet is the one we want to delete it's a bit of a risk. `active` is actually a state that gets sent over the websocket so the backend knows to send us more data. It'd be much safer to pass the subnet's id into this component.